micro-rdk/src/common/data_collector.rs

```diff
@@ -1,5 +1,5 @@
 use std::fmt::Display;
-use std::time::Duration;
+use std::time::{Duration, Instant};
```

Comment:
I believe so, but I'll double-check to confirm the behavior of `Instant` is as expected. I think it's worth noting that, since we're using elapsed, it doesn't really matter whether `Instant::now` returns zero.

Reply:
So, for UNIX systems, `Instant::now` uses [clock_gettime()](https://doc.rust-lang.org/nightly/src/std/time.rs.html#113). And I believe for the xtensa targets, UNIX convention is mostly used because ESP-IDF defines many of the same kernel functions. Following the thread from [here](https://github.com/espressif/esp-idf/blob/release/v4.4/components/newlib/time.c#L269) through [here](https://github.com/espressif/esp-idf/blob/release/v4.4/components/newlib/port/esp_time_impl.c#L90), it would seem ESP-IDF uses the RTC timer (which advances during light sleep) in its implementation of `clock_gettime`, meaning light sleep should probably not interfere with the behavior of Instant

Reply:
OK, I'm reasonably convinced. I wouldn't mind seeing a test added that puts the ESP32 into light sleep and validates that time advances plausibly on wakeup, but I'd also be willing to defer that into a ticket. Could you please file one? I just want to ensure that if the behavior changes on us, we find out in our tests, not the field.